cpp/src/slice2cpp/Gen.cpp

```diff
@@ -908,20 +950,33 @@ Slice::Gen::generate(const UnitPtr& p)
     {
         normalizeMetaData(p, true);
```

Comment:
I would maybe add a bigger comment above all the visitors describing the order of their usage. Currently I don't think the comments are easy to understand on their own.